src/Microsoft.ML.SearchSpace/Schema/SchemaBase.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.ML.SearchSpace.Schema
+{
+    public abstract class SchemaBase
+    {
+        [JsonProperty(PropertyName = "schema_type", Required = Required.Always)]
+        public abstract SchemaType SchemaType { get; }
```

Comment:
What is `SchemaType` used for? I don't see any callers of it.

Reply:
It's for serializing/deserializing `SearchSpace`. Each option (choice/numeric) will have its own schema and schema will be in response to all serializing work. We are still working on this so currently no caller for it. But essentially it's something like this

```
SearchSpace <==> SearchSpace Schema <==> Json/Xml
```

By doing so we pull out the serializing work into a separate class, which will be helpful if there are multiple formats to convert to. (for example, aml sweeping or from an old version of search space json)

Reply:
If this is just a separate class for serializing, I don't think it should be public.

Reply:
### Resolved